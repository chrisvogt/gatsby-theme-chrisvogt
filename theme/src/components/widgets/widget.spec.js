import React from 'react'
import renderer from 'react-test-renderer'
import Widget from './widget'

describe('Widget', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Widget>Test</Widget>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('sets the widget id attribute when provided', () => {
    const id = 'fake-widget'
    const testRenderer = renderer.create(<Widget id={id}>Test</Widget>)
    const testInstance = testRenderer.root
    expect(testInstance.findByType('section').props.id).toEqual(id)
  })
})
