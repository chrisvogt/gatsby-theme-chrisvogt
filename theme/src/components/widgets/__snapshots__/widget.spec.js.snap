// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget matches the snapshot 1`] = `
<section
  className="css-4jhz55-Widget"
>
  Test
</section>
`;
